Python/correlation/test/test_correlation.py

```diff
@@ -0,0 +1,142 @@
+"""
+Provides test to visualize results, and check how accurate the result is when varying
+sample frequency, pulse length, position of hydrophones and pinger, and window length.
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+import pytest
+
+import Python.correlation.correlation as correl
+import Python.correlation.tdoa as tdoa
+
+import Python.signal_generation.noise as sg_noise
+import Python.signal_generation.positioning as sg_pos
+import Python.signal_generation.receiver as sg_rec
+import Python.signal_generation.source as sg_src
+
+#@pytest.mark.plot
+def test_plot_correlation():
+    # Set variables:
+    sample_frequency = 500.0  # [kHz]
+    pulse_lenth = 5  # [ms]
+    signal_length = 10000  # [samples]
+
+
+    positions = np.array(
+        [
+            sg_pos.Position(x=0.0, y=0.0, z=0.0),
+            sg_pos.Position(x=1.83, y=0.0, z=0.0),
+            sg_pos.Position(x=0.0, y=0.91, z=0.91),
+            sg_pos.Position(x=1.83, y=0.91, z=0.0),
+            sg_pos.Position(x=1.0, y=0.0, z=0.91),
+        ]
+    )
+
+    hydro_array = sg_rec.HydrophoneArray(
+        positions=positions,
+    )
+
+    pinger_position_one = sg_pos.Position(
+        x=4.0,
+        y=0.0,
+        z=0.0,
+    )
+
+    pinger = [
+        sg_src.Pinger(
+            frequency=25.0,  # [kHz]
+            sampling_frequency=sample_frequency,  # [kHz]
+            period=30.0,
+            pulse_length=pulse_lenth,
+            use_window=True,
+            position=pinger_position_one,
+        ),
+    ]
+
+    resulting_signals = sg_rec.generate_signals(
+        sources=pinger,
```

Comment:
```suggestion
        sources=np.array([pinger]),
```
The function expects a numpy array.